src/mesh/interpolation.cxx

```diff
@@ -63,161 +64,179 @@ const Field3D interp_to(const Field3D &var, CELL_LOC loc)
+
+            if ((location == CELL_CENTRE) && (loc == CELL_YLOW)) {
+              // Producing a stencil centred around a lower Y value
+              s.pp = s.p;
+              s.p  = s.c;
+            } else if(location == CELL_YLOW) {
+              // Stencil centred around a cell centre
+              s.mm = s.m;
+              s.m  = s.c;
+            }
+
+            result[i] = interp(s);
+          }
+          } else {
+            // Only one guard cell, so no pp or mm values
+            s.pp = nan("");
+            s.mm = nan("");
+            throw BoutException("Cannot interpolate in Y direction\n"
+                                " - Not enough boundary cells\n"
+                                " - at least 2 are needed!");
             for(const auto &i : result.region(RGN_NOY)) {
-	      // Set stencils
-	      s.c = var_fa[i];
-	      s.p = var_fa[i.yp()];
-	      s.m = var_fa[i.ym()];
-	      
-	      if ((location == CELL_CENTRE) && (loc == CELL_YLOW)) {
-		// Producing a stencil centred around a lower Y value
-		s.pp = s.p;
-		s.p  = s.c;
-	      } else if(location == CELL_YLOW) {
-		// Stencil centred around a cell centre
-		s.mm = s.m;
-		s.m  = s.c;
-	      }
-	      
-	      result[i] = interp(s);
-	    }
-	  }
-	}
-	break;
-	// Need to communicate in Y
+              // Set stencils
+              s.c = var_fa[i];
+              s.p = var_fa[i.yp()];
+              s.m = var_fa[i.ym()];
+
+              if ((location == CELL_CENTRE) && (loc == CELL_YLOW)) {
+                // Producing a stencil centred around a lower Y value
+                s.pp = s.p;
+                s.p  = s.c;
+              } else if(location == CELL_YLOW) {
+                // Stencil centred around a cell centre
+                s.mm = s.m;
+                s.m  = s.c;
+              } else {
+                // We should never be here
+                throw BoutException("Please report this bug!");
+              }
+
+              result[i] = interp(s);
+            }
+          }
+        }
+        break;
+        // Need to communicate in Y
       }
       case CELL_ZLOW: {
         for(const auto &i : result.region(RGN_NOZ)) {
-	  s.c = var[i];
-	  s.p = var[i.zp()];
-	  s.m = var[i.zm()];
-	  s.pp = var[i.offset(0,0,2)];
-	  s.mm = var[i.offset(0,0,-2)];
-	  
-	  result[i] = interp(s);
-	}
-	break;
+          s.c = var[i];
+          s.p = var[i.zp()];
+          s.m = var[i.zm()];
+          s.pp = var[i.offset(0,0,2)];
+          s.mm = var[i.offset(0,0,-2)];
+
+          if ((location == CELL_CENTRE) && (loc == CELL_ZLOW)) {
+            // Producing a stencil centred around a lower Z value
+            s.pp = s.p;
+            s.p  = s.c;
+          } else if(location == CELL_ZLOW) {
+            // Stencil centred around a cell centre
+            s.mm = s.m;
+            s.m  = s.c;
+          } else {
+            // We should never be here
+            throw BoutException("Please report this bug!");
```

Comment:
I have no clue why it would be there.

I tried to change the code as little as possible - but if it does take nether of the first two cases, something is wrong.

If the code would get there, that would need some diagnoses, on why we did get here ...